### PR TITLE
feat(dispatcher): add event relay dispatcher and receiver modules

### PR DESCRIPTION
**Pull request by wangrong1069**
Add event_relay_dispatcher and event_relay_receiver for GIO-based DBus event relay, and enable testing in top-level CMakeLists.

新增事件中继调度器与接收器模块，支持通过 DBus 进行事件转发。

Log: 新增事件中继调度器与接收器模块
PMS: BUG-370779
Influence: 新增 dispatcher 事件中继功能，支持 DBus 方式转发事件，并在顶层构建系统中启用 ctest。

## Summary by Sourcery

Introduce DBus event relay components that broadcast dispatcher events to connected clients and enable their tests through the top-level build.

New Features:
- Add GIO/DBus-based event relay dispatcher and receiver modules for broadcasting events over dynamically created channels.
- Support receiver-side acquisition of event channels and protocol identifiers through D-Bus file-descriptor passing.

Enhancements:
- Provide channel limits, nonblocking socket communication, and automatic cleanup of disconnected or slow relay clients.

Build:
- Add GIO and GIO Unix dependencies to the dispatcher library.
- Enable top-level test configuration and CTest discovery across component test directories.

Tests:
- Add relay dispatcher tests covering channel creation, limits, broadcasting, cleanup, message boundaries, receiver failure handling, and end-to-end delivery.

### CODE REVIEW

File: src/dispatcher/event_relay_receiver.c [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        if (errno == EAGAIN || errno == EWOULDBLOCK)
+            return EVENT_RECEIVE_WOULD_BLOCK;
+        g_debug("relay: recv() failed: %s", strerror(errno));
+        return EVENT_RECEIVE_ERROR;
+    }
+
+    if (ret == 0)
+        return EVENT_RECEIVE_DISCONNECTED;
+
+    return EVENT_RECEIVE_OK;
```
[the remark this answers is not in the document]
Comment (wangrong1069) [reply src/dispatcher/event_relay_receiver.c:174]: 客户端和服务器使用相同大小的 buf

File: src/dispatcher/event_relay_dispatcher.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        ssize_t ret = send(fd, buf, len, MSG_NOSIGNAL);
+        if (ret < 0) {
+            if (errno == EAGAIN || errno == EWOULDBLOCK) {
+                g_debug("relay: slow client fd %d, kicking", fd);
+                close(fd);
+                g_ptr_array_remove_index_fast(dispatcher->socket_list, i);
+            } else if (errno == EPIPE || errno == ECONNRESET) {
+                g_debug("relay: broken connection fd %d", fd);
+                close(fd);
+                g_ptr_array_remove_index_fast(dispatcher->socket_list, i);
```
[the remark this answers is not in the document]
Comment (wangrong1069) [reply src/dispatcher/event_relay_dispatcher.c:119]: 已修复

File: CMakeLists.txt [hunk 3]
```diff
@@ -9,5 +9,10 @@ if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
     set(CMAKE_INSTALL_PREFIX /usr)
 endif ()
 
+# Let `ctest` from the top-level build dir traverse into component test
+# dirs (each component registers its tests via its own ENABLE_TESTING).
+# Must precede add_subdirectory: testing support is only wired into
+# subdirectories added after this call.
+enable_testing()
```
[the remark this answers is not in the document]
Comment (wangrong1069) [reply CMakeLists.txt:21]: 已修复

File: src/dispatcher/event_relay_receiver.c [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    g_variant_unref(result);
+
+    if (fd_list == NULL || g_unix_fd_list_get_length(fd_list) < 1) {
+        g_warning("relay: GetEventChannel returned no file descriptor");
+        g_clear_object(&fd_list);
+        goto fail;
+    }
+
+    g_clear_error(&error);
+    gint fd = g_unix_fd_list_get(fd_list, 0, &error);
```
[the remark this answers is not in the document]
Comment (wangrong1069) [reply src/dispatcher/event_relay_receiver.c]: 已修复

File: src/dispatcher/event_relay_dispatcher.c [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            } else {
+                g_warning("relay: send() to fd %d failed: %s",
+                          fd, strerror(errno));
+            }
```
[the remark this answers is not in the document]
Comment (wangrong1069) [reply src/dispatcher/event_relay_dispatcher.c:126]: 已修复

File: src/dispatcher/tests/test_relay_dispatcher.c [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+    /* Parent: child should exit 0 (got NULL, possibly via g_warning abort) */
+    int status;
+    waitpid(pid, &status, 0);
+
+    /* The child may exit 0 (if g_warning wasn't fatal for some reason) or be
+     * killed by a signal from the fatal-warning handler (SIGABRT or SIGTRAP
+     * depending on the GLib backend). Accept either outcome. */
+    g_assert_true(WIFEXITED(status) || WIFSIGNALED(status));
+    if (WIFEXITED(status))
+        g_assert_cmpint(WEXITSTATUS(status), ==, 0);
```
[the remark this answers is not in the document]
Comment (wangrong1069) [reply src/dispatcher/tests/test_relay_dispatcher.c]: 已修复

**Review by max-lvs — APPROVED** [review]
